GPU/Common/FramebufferCommon.cpp

```diff
@@ -1352,9 +1352,10 @@ void FramebufferManagerCommon::FindTransferFramebuffers(VirtualFramebuffer *&dst
 			const u32 byteOffset = dstBasePtr - vfb_address;
 			const u32 byteStride = dstStride * bpp;
 			const u32 yOffset = byteOffset / byteStride;
+
 			// Some games use mismatching bitdepths.  But make sure the stride matches.
 			// If it doesn't, generally this means we detected the framebuffer with too large a height.
-			bool match = yOffset < dstYOffset;
+			bool match = yOffset < dstYOffset && yOffset < vfb->height;
```

Comment:
Hmm, what are the values here?  We check `dstBasePtr < vfb_address + vfb_size`, which should mean the dstBasePtr isn't beyond the size (and therefore height) of the vfb.

If this is preventing a match, doesn't it mean that something is wrong with that check or the size calculation somewhere?

-[Unknown]

Reply:
The values when it happens are yOffset = 3072 (0xc00) and vfb->height = 272. The reason that yOffset can get so big is that in this particular copy the stride is 32 (or maybe 64, don't remember) and if the framebuffer we're trying to match to (which might no longer be valid, not sure) is large and the stride is small, byteOffset/byteStride can get large.

It's a bit tricky to trigger, happens occasionally when driving around some tracks in Burnout, I believe it's related to the sun readbacks for the flare effect.

Reply:
Oh, duh, the yOffset is based on a different stride than vfb_size.  Got it.

I've seen games use weird transfer strides, though.  For example one game used a stride of 8, or something similar, it seemed like the developer thought it'd go faster copying by 8 pixels.  Anyway, I can't remember what game it was... I feel like it was God of War, but not sure.

-[Unknown]

Reply:
Well in that case, the match will fail and the copy will fall back to a CPU copy, which at least in the God of War case would do the right thing anyway because IIRC it's using it to upload textures to VRAM. Anyway I don't think the change will break anything because any copies it now would not detect would generate bad copy operations anyway due to the yoffset being too large...  but this stuff might be in need of a bit of deeper thinking :)